R/scripts/filter_MC.R

```diff
@@ -9,20 +9,21 @@ suppressMessages(library(magrittr))
 suppressMessages(library(xts))
 suppressMessages(library(reticulate))
 suppressMessages(library(truncnorm))
+suppressMessages(library(parallel))
 # suppressMessages(library(flock))
 options(warn=1)
 
 option_list = list(
-  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("CONFIG_PATH"), type='character', help="path to the config file"),
-  optparse::make_option(c("-s", "--scenarios"), action="store", default='all', type='character', help="name of the intervention to run, or 'all' to run all of them"),
-  optparse::make_option(c("-d", "--deathrates"), action="store", default='all', type='character', help="name of the death scenarios to run, or 'all' to run all of them"),
-  optparse::make_option(c("-j", "--jobs"), action="store", default="8", type='integer', help="Number of jobs to run in parallel"),
-  optparse::make_option(c("-k", "--simulations_per_slot"), action="store", default=NA, type='integer', help = "number of simulations to run for this slot"),
-  optparse::make_option(c("-n", "--number_of_simulations"), action="store", default="1", type='integer', help = "number of slots to run"),
-  optparse::make_option(c("-i", "--this_slot"), action="store", default="1", type='integer', help = "id of this slot"),
+  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("COVID_CONFIG_PATH", Sys.getenv("CONFIG_PATH")), type='character', help="path to the config file"),
+  optparse::make_option(c("-s", "--scenarios"), action="store", default=Sys.getenv("COVID_SCENARIOS", 'all'), type='character', help="name of the intervention to run, or 'all' to run all of them"),
+  optparse::make_option(c("-d", "--deathrates"), action="store", default=Sys.getenv("COVID_DEATHRATES", 'all'), type='character', help="name of the death scenarios to run, or 'all' to run all of them"),
+  optparse::make_option(c("-j", "--jobs"), action="store", default=Sys.getenv("COVID_NJOBS", parallel::detectCores()), type='integer', help="Number of jobs to run in parallel"),
+  optparse::make_option(c("-k", "--simulations_per_slot"), action="store", default=Sys.getenv("COVID_SIMULATIONS_PER_SLOT"), type='integer', help = "number of simulations to run for this slot"),
+  optparse::make_option(c("-n", "--number_of_simulations"), action="store", default=Sys.getenv("COVID_NSIMULATIONS", 1), type='integer', help = "number of slots to run"),
```

Comment:
suggestions for dealing with the confusing simulations/slot nomenclature? maybe i could tackle it here

Reply:
i'm planning on tackling this as part of filename fix, but aprt of that can be done immediately:
 - slots are normally called chains (n->N,i->n)
 - simulations per slot is normally called iterations (k->K,new k)
 - maybe go ahead and add an unused block (new B,new b)
 - I think using different letters for # of x and index inside x is weird, and we should just do lowercase and capital
open to other suggestions

Reply:
so:
- `COVID_NSIMULATIONS` -> `COVID_NCHAINS`
- `COVID_SIMULATIONS_PER_SLOT` -> `COVID_ITERATIONS`

this should apply only to the inference jobs? what about the existing seir/hosp code and the params they take?

Reply:
that's a good point.  i think we should continue calling them simulations. So maybe COVID_NSIMULATIONS and COVID_FIRST_SIMULATION.